ci/taos/plugins-staging/pr-nnstreamer-apptest.sh

```diff
@@ -0,0 +1,115 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+##
+# @file pr-nnstreamer-apptest.sh
+# @brief Check nnstreamer sample app working well.
+# @see      https://github.com/nnsuite/TAOS-CI
+# @author   Sewon Oh <sewon.oh@samsung.com>
+
+# @brief Set variable to run
+export nnstreamer_apptest=1
+
+# @brief [MODULE] TAOS/pr-nnstreamer-apptest-trigger-queue
+function pr-nnstreamer-apptest-trigger-queue(){
+    message="Trigger: queued. There are other build jobs and we need to wait.. The commit number is $input_commit."
+    cibot_pr_report $TOKEN "pending" "TAOS/pr-nnstreamer-apptest" "$message" "${CISERVER}${PRJ_REPO_UPSTREAM}/ci/${dir_commit}/" "$GITHUB_WEBHOOK_API/statuses/$input_commit"
+}
+
+# @brief [MODULE] TAOS/pr-nnstreamer-apptest-trigger-run
+function pr-nnstreamer-apptest-trigger-run(){
+    echo "[DEBUG] Starting CI trigger to run nnstreamer sample app actually."
+    message="Trigger: running. The commit number is $input_commit."
+    cibot_pr_report $TOKEN "pending" "TAOS/pr-nnstreamer-apptest" "$message" "${CISERVER}${PRJ_REPO_UPSTREAM}/ci/${dir_commit}/" "$GITHUB_WEBHOOK_API/statuses/$input_commit"
+}
+
+# @brief [MODULE] TAOS/pr-nnstreamer-apptest
+function pr-nnstreamer-apptest() {
+    echo "########################################################################################"
+    echo "[MODULE] TAOS/pr-nnstreamer-apptest: Starting sample app test"
+
+    check_dependency cmake
+    check_dependency make
+    check_dependency python
+
+    # Set-up environment variables.
+    export NNST_ROOT="${dir_ci}/${dir_commit}/${PRJ_REPO_OWNER}"
+    export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$NNST_ROOT/lib
+    export GST_PLUGIN_PATH=$GST_PLUGIN_PATH:$NNST_ROOT/lib
+
+    # build and install nnstreamer library
+    cd ${NNST_ROOT}
+    mkdir build
+    cd build
+    rm -rf *
+    cmake -DCMAKE_INSTALL_PREFIX=${NNST_ROOT} \
+    -DINCLUDE_INSTALL_DIR=${NNST_ROOT}/include \
+    -DGST_INSTALL_DIR=${NNST_ROOT}/lib ..
+    make install
+    cd ..
+
+    # Set-up testing environment.
+    mkdir bin
+    cp build/nnstreamer_example/example_filter/nnstreamer_example_filter bin/
+    cp nnstreamer_example_filter/example_filter/nnstreamer_example_filter.py bin/
+    cp build/nnstreamer_example/example_cam/nnstreamer_example_cam bin/
+    cp build/nnstreamer_example/example_sink/nnstreamer_sink_example bin/
+    cp build/nnstreamer_example/example_sink/nnstreamer_sink_example_play bin/
+    rm -rf build
+    cd bin
+
+    # Download tensorflow-lite model file and labels.
+    mkdir tflite_model; cd tflite_model
+    wget https://github.com/nnsuite/testcases/tree/master/DeepLearningModels/tensorflow-lite/Mobilenet_1.0_224_quant/mobilenet_v1_1.0_224_quant.tflite
+    tar xvzf ./mobilenet_v1_1.0_224_quant.tgz
+    wget https://raw.githubusercontent.com/nnsuite/testcases/master/DeepLearningModels/tensorflow-lite/Mobilenet_1.0_224_quant/labels.txt
+    cd ..
```

Comment:
How about we try to append a check routine with '$?' variable if  the download  result is normal?

Reply:
Thats good idea.